src/common/snippets/src/op/subgraph.cpp

```diff
@@ -382,6 +384,9 @@ void Subgraph::data_flow_transformations(const BlockedShapeVector& blocked_input
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::op::data_flow_transformations")
 
     ov::snippets::pass::Manager manager;
+    // GroupNormalizationDecomposition should be before canonicalization(rankNorm) as scale/bias shape is C and need special process.
+    if (config.m_has_domain_sensitive_ops)
+        manager.register_pass<snippets::pass::GroupNormalizationDecomposition>();
```

Comment:
I think we should register this decomposition in [CommonOptimizations](https://github.com/openvinotoolkit/openvino/blob/5e7e51dc77822ab6cb9e45c7208100c57bec4104/src/common/snippets/src/pass/common_optimizations.cpp#L32) as `FQDecomposition`. Could you check please? You also need to add `Reduce` op to [this](https://github.com/openvinotoolkit/openvino/blob/5e7e51dc77822ab6cb9e45c7208100c57bec4104/src/common/snippets/src/op/subgraph.cpp#L271) method to avoid extra Constant with axes extraction in the following pass `ExtractConstants`

Reply:
If decomposed in common optimization, it will go common [TokenizeSnippets](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/pass/collapse_subgraph.cpp), that will fuse more ops into GroupNorm. GroupNorm is fixed to parallel on N*group level, which is not optimal for other subgraph. As discussed, we can decomposed GN in common optimization after parallel is moved into snippets with more flexible scheduling. Currently we have unified parallel scheduling.
Anyway, if some light-weight post-ops exist after GN, we can fused into GN in GNTokenzation.

Reply:
I meant common optimizations - snippets common optimizations (I attached the link before - take a [look](https://github.com/openvinotoolkit/openvino/blob/5e7e51dc77822ab6cb9e45c7208100c57bec4104/src/common/snippets/src/pass/common_optimizations.cpp#L32) please). These optimizations are called after all [tokenization](https://github.com/openvinotoolkit/openvino/blob/5e7e51dc77822ab6cb9e45c7208100c57bec4104/src/common/snippets/src/pass/tokenization.cpp#L85) passes. Could you please take a look one more time?

Reply:
done, but after this change, there are several single layer tests fail. Do you is it right place [here](https://github.com/openvinotoolkit/openvino/pull/21751/commits/ebf76d1b06c969b978c9cd841cc7519a7d5a667a#diff-e8ddefdc994e3b2a6ac64dbf5d884002b0e361386c164c3bf1602998caf33b51R54) , I also add reduce to [ExtractConstants](https://github.com/openvinotoolkit/openvino/pull/21751/commits/ebf76d1b06c969b978c9cd841cc7519a7d5a667a#diff-49fb24bc86fb62b627288d2d39e027abb1e6d8740c46a1351026dc7f6b45d550R283). analyzing the reason.

Reply:
Removed to snippets common optimization. This subgraph should be executed in domain_sensitive manner. If decompose in common optimization. This info is lost as it does not have GN anymore. As reshape impact domain_optimization, I add reshape to  [is_domain_sensitive_op](https://github.com/openvinotoolkit/openvino/pull/21751/commits/cfd33302a2b4380cecf8de872feffeba81c1c57b#diff-49fb24bc86fb62b627288d2d39e027abb1e6d8740c46a1351026dc7f6b45d550R82) , then in[ Subgraph::init_config()](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/op/subgraph.cpp#L83), this info is kept. Welcome comment if you have any better idea. Thanks Alexandra!